### PR TITLE
feat: knowledge graph agent

### LINKED ISSUE #489: [Feature Request] Knowledge graph agent
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [X] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

for knowledge graph retriever, we need to get the relationship between different entities, this can be done by using a LLM-based agent.

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

for knowledge graph retriever, we need to get the relationship between different entities, this can be done by using a LLM-based agent.

This agent would be able to give structured string content back, this string content will be validated further to be put into a `GraphElement`

## Motivation and Context

 `close #489 ` 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/agents/knowledge_graph_agent.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+from camel.storages.graph_storages.graph_element import (
+    GraphElement,
+    Node,
+    Relationship,
+)
+from camel.types import ModelType, RoleType
+
+text_prompt = """
```
Comment (zechengz) [inline camel/agents/knowledge_graph_agent.py:28]: Can we move this prompt to somewhere else such as the prompts?

File: camel/agents/knowledge_graph_agent.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+Determine relationships between these entities and represent them as directed relationships.
+Provide the extracted nodes and relationships in the specified format below.
+Example for you:
+
+Example Content:
+"John works at XYZ Corporation. He is a software engineer. The company is located in New York City."
+
+Expected Output:
+
+Nodes:
```
Comment (zechengz) [inline camel/agents/knowledge_graph_agent.py:63]: For KG we usually have Entity and Relation. Maybe we can change node to entity and relationship to Relation.

**Review by zechengz — COMMENTED** [review]
Left some comments

File: camel/agents/knowledge_graph_agent.py (hunk 2 above)
Comment (Wendong-Fan) [reply camel/agents/knowledge_graph_agent.py:63]: In Neo4j they use `Node` and `Relationship` as the screenshot below, how about keep the naming align with Neo4j?

Screenshot 2024-05-23 at 01 15 11

File: camel/agents/knowledge_graph_agent.py (hunk 1 above)
Comment (Wendong-Fan) [reply camel/agents/knowledge_graph_agent.py:28]: Yeah I agree with you we can manage prompts better, but currently we also have other agents putting prompt directly in the main code file, how about leave it for now, later we can do uniformed enhance for all prompts in camel?

**Review by zechengz — APPROVED** [review]

**Pull request merged**